RyuSocks/Auth/UsernameAndPassword.cs

```diff
@@ -25,19 +28,48 @@ namespace RyuSocks.Auth
     [AuthMethodImpl(0x02)]
     public class UsernameAndPassword : IProxyAuth
     {
+        private readonly Dictionary<string, string> _database;
+
+        public UsernameAndPassword(Dictionary<string, string> database)
+        {
+            _database = database;
+        }
+
         public bool Authenticate(ReadOnlySpan<byte> incomingPacket, out ReadOnlySpan<byte> outgoingPacket)
         {
-            throw new NotImplementedException();
+            UsernameAndPasswordRequest requestPacket = new(incomingPacket.ToArray());
+
+            requestPacket.Validate();
```

Comment:
You are assuming that only the server calls this method. But as the interface specifies, it is supposed to be called by both clients and servers.

Could you make sure to handle both cases in here? Feel free to just add a bool that gets set to true/false when the server-side init method gets called.

Reply:
We talked about this and did not come to a conclusion.
I require more input.